### PR TITLE
Add membership number circle

### PR DESCRIPTION
**Pull request by rarira**
## Specify project
Both server and client side related

## Description

* Added counting badge (`+number`) over the thumbnail images(max 4).
* Updated server side codes
   1. When a user create new message in a channel, it also updates `updatedAt` field of the `Membership`
   ```
   \\\ added to resolvers/Message/mutation.ts
   export const createMessage = mutationField('createMessage',
    ........

        await prisma.membership.update({
      data: {
        updatedAt: created.createdAt,
      },
      where: {
        userId_channelId: {
          userId,
          channelId,
        },
      },
    });
  ```

   2. `Membership` model returns users sorted by `updatedAt` to show the name and thumbnail of the latest active(sent the message) user first.

    ```
    \\\ models/channel.ts

      resolve: ({id}, {excludeMe}, {prisma, userId}) => {
        if (excludeMe)
          return prisma.membership.findMany({
            where: {
              channel: {id},
              user: {
                id: {not: userId ?? undefined},
              },
            },
            include: {user: true},
            orderBy: [
              {
                updatedAt: 'desc',
              },
            ],
          });

        return prisma.membership.findMany({
          where: {channel: {id}},
          include: {user: true},
          orderBy: [
            {
              updatedAt: 'desc',
            },
          ],
        });
      }

    ```

   3. Updated subscription 'onMessage' accordingly for real-time UI update
   

   ```
    \\\\ components/navigations/MainStackNavigator/index.tsx

  subscription MainStackNavigatorOnMessageSubscription($deviceKey: String!) {
    onMessage(deviceKey: $deviceKey) {
      id
      imageUrls
      channel {
        id
        lastMessage {
          id
          messageType
          text
          imageUrls
          fileUrls
          createdAt
        }

       // added lines
        memberships(excludeMe: true) {
          user {
            name
            nickname
            thumbURL
            photoURL
          }
        }

      }
      sender {
        id
        name
        nickname
      }
      createdAt
      ...MessageListItem_message
    }
  }

   ```

## Related Issues

#315 1.Show all the users in the channel.

## Tests
 I'm sorry, writing test codes from the scratch is too hard for me now, I'll try next time

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn lint && yarn tsc`
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] I am willing to follow-up on review comments in a timely manner.

[wehack2021]-[cloud]

### CODE REVIEW

**Review by hyochan — APPROVED** [review]
great

You got a very good understanding of the requested feature. LGTM!

**Comment by geoseong** [comment]
image
LGTM!

and I think it would be good to put any comment about Test code like,
```
## Tests
ex 1) nothing to do
ex 2) we will do next feature
ex 3) couldn't do. it's hard for me so far
```

if we continuously skip the test codes and writing the description without any comments about "Test" section, we would forget to implement any test codes.

Writing test codes is hard for me too but we can try to keep thinking of writing test codes. 👍😃

**Review by geoseong — APPROVED** [review]

File: server/src/models/Channel.ts [hunk 1]
```diff
@@ -84,11 +84,21 @@ export const Channel = objectType({
               },
             },
             include: {user: true},
+            orderBy: [
+              {
+                updatedAt: 'desc',
+              },
+            ],
```
Comment (0916dhkim) [inline server/src/models/Channel.ts:91]: Nice quality-of-life addition :100:

File: server/src/resolvers/Message/mutation.ts [hunk 2]
```diff
@@ -62,6 +62,18 @@ export const createMessage = mutationField('createMessage', {
...
+      data: {
+        updatedAt: created.createdAt,
+      },
+      where: {
+        userId_channelId: {
+          userId,
+          channelId,
+        },
+      },
+    });
```
Comment (0916dhkim) [inline server/src/resolvers/Message/mutation.ts:75]: I think we can combine `prisma.channel.update` and `prisma.membership.update` into a single update using Prisma's "nested writes" feature.
Reference: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes

**Review by 0916dhkim — APPROVED** [review]
Looks good to me. I have a small suggestion below :point_down:

File: server/src/resolvers/Message/mutation.ts (hunk 2 above)
Comment (rarira) [reply server/src/resolvers/Message/mutation.ts:75]: thanks for your suggestion, I'll read and try ASAP :)

Comment (rarira) [reply server/src/resolvers/Message/mutation.ts]: Updated!

**Pull request merged**